### ISSUE TITLE
motion sensor question

### ISSUE THREAD

**Issue by kefir135** [csuermann/node-red-contrib-virtual-smart-home#177]
Hi, do I need a PRO plan to use motion sensor?
"Normal" devices work ok - sending messages from Alexa to NODE-RED, but after adding a motion sensor, in Alexa, I had "waiting for virtual smart home", and afeter sending this message:
{
    "detectionState": "DETECTED",
    "source": "device",
    "directive": "OverrideLocalState",
    "name": "red motion",
    "type": "MOTION_SENSOR"
}
to the node (as payload), device in Alexa wrote "There was a problem" and I can't even open the device in Alexa app.

Basically, Alexa --> Node Red works ok, but I need it the oposite way - NoDe-Red --> Alexa.

Best,
Rafal

**Comment by csuermann** [comment]
Hi @kefir135,

Thanks for reaching out! The PRO plan is needed if you want Alexa (i.e. Alexa app) to retrieve the state of your VSH devices. I believe using a motion sensor to trigger a routine should still be possible without PRO, though.

**Comment by kefir135** [comment]
Ho Cornelius,

1. I just upgraded to PRO - but it is visible at the connection tab, but
not on device, see screenshots, is this correct behavior?
2. Motion sensor still not functional in Alexa App :-(
[image: image.png]
[image: image.png]

Best regards
Kefir135

sob., 7 sty 2023 o 11:06 Cornelius Suermann ***@***.***>
napisał(a):

**Comment by csuermann** [comment]
Hi @kefir135,

have you retried restarting your flows or Node-RED altogether? There seems to be an issue with correctly displaying the status of the plan on the vsh-device page. Restarting might fix it.

**Comment by kefir135** [comment]
Thanks!
Restarting node-red fixed the status.
Still, all Alexa sees from thos motion sensor is "There was a problem" :-(



śr., 11 sty 2023 o 12:51 Cornelius Suermann ***@***.***>
napisał(a):

**Comment by csuermann** [comment]
Thanks for reporting back that restarting fixed the display issue. Did you enable the "retrievable" option for your motion sensor vsh-device in Node-RED?

**Comment by kefir135** [comment]
I did. With the new device. Alexa discovered it, for one second there was
"no motion detected" on the device screen in alexa app, then the device
screen went blank, and "There was a problem" appeared on the devices list.


śr., 11 sty 2023 o 15:24 Cornelius Suermann ***@***.***>
napisał(a):

**Comment by csuermann** [comment]
Could you kindly switch on the debug option for your vsh-connection and share the output of the Node-RED CLI with me (vsh@csuermann.de) from the moment of starting NR to the moment where the problem occurs? I'll then try to analyze what's going on.

**Comment by kefir135** [comment]
Hi, is this what you were asking for?



***@***.***:~ $ node-red-start

Start Node-RED

Once Node-RED has started, point a browser at http://192.168.50.80:1880
On Pi Node-RED works better with the Firefox or Chrome browser

Use   node-red-stop                          to stop Node-RED
Use   node-red-start                         to start Node-RED again
Use   node-red-log                           to view the recent log output
Use   sudo systemctl enable nodered.service  to autostart Node-RED at every
boot
Use   sudo systemctl disable nodered.service to disable autostart on boot

To find more nodes and example flows - go to http://flows.nodered.org

Starting as a systemd service.
15 Jan 19:05:05 - [info]
Welcome to Node-RED
===================
15 Jan 19:05:05 - [info] Node-RED version: v3.0.2
15 Jan 19:05:05 - [info] Node.js  version: v16.19.0
15 Jan 19:05:05 - [info] Linux 5.15.76-v8+ arm64 LE
15 Jan 19:05:06 - [info] Loading palette nodes
15 Jan 19:05:08 - [info] Settings file  : /home/pi/.node-red/settings.js
15 Jan 19:05:08 - [info] Context store  : 'default' [module=memory]
15 Jan 19:05:08 - [info] User directory : /home/pi/.node-red
15 Jan 19:05:08 - [warn] Projects disabled :
editorTheme.projects.enabled=false
15 Jan 19:05:08 - [info] Flows file     : /home/pi/.node-red/flows.json
15 Jan 19:05:08 - [info] Server now running at http://127.0.0.1:1880/
15 Jan 19:05:08 - [warn]
---------------------------------------------------------------------
Your flow credentials file is encrypted using a system-generated key.
If the system-generated key is lost for any reason, your credentials
file will not be recoverable, you will have to delete it and re-enter
your credentials.
You should set your own key using the 'credentialSecret' option in
your settings file. Node-RED will then re-encrypt your credentials
file using your chosen key the next time you deploy a change.
---------------------------------------------------------------------
15 Jan 19:05:09 - [info] Starting flows
15 Jan 19:05:09 - [info] Started flows
15 Jan 19:05:09 - [info] [mqtt-broker:mosquitto] Connected to broker:
mqtt://localhost:1883
15 Jan 19:05:09 - [info] ***@***.***
Attempting MQTT connection:
a1pv0eq8s016ut-ats.iot.eu-west-1.amazonaws.com:8883 (clientId:
vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf)
15 Jan 19:05:09 - [info] ***@***.*** MQTT
subscribe to topics:
["$aws/things/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/shadow/get/accepted","vsh/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/+/directive","vsh/service","vsh/version/3.0.7/+","vsh/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/service"]
15 Jan 19:05:09 - [info] ***@***.*** MQTT
connecting to a1pv0eq8s016ut-ats.iot.eu-west-1.amazonaws.com:8883
15 Jan 19:05:12 - [info] ***@***.*** MQTT
publish to topic
vsh/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/requestConfig:
{"vshVersion":"3.0.7"}
15 Jan 19:05:12 - [info] ***@***.*** MQTT
message received on topic
vsh/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/service:
{"operation":"overrideConfig","userIdToken":"amzn1.account.AGMCT45ZEN4ZTKV7632BTOCNMGAQ#00a5e4d44ba0cecc2296127d2e2ab6043e994abc","allowedDeviceCount":200,"plan":"pro","msgRateLimiter":{"profiles":{"DEFAULT":{"maxConcurrent":1,"minTime":1000,"highWater":0,"strategy":"BLOCK","penalty":30000,"reservoir":60,"reservoirIncreaseInterval":60000,"reservoirIncreaseAmount":1,"reservoirIncreaseMaximum":60},"PHYSICAL_INTERACTION_DEFAULT":{"maxConcurrent":1,"minTime":2500,"highWater":1,"strategy":"LEAK","reservoir":30,"reservoirIncreaseInterval":900000,"reservoirIncreaseAmount":1,"reservoirIncreaseMaximum":15},"VOICE_INTERACTION_DEFAULT":{"maxConcurrent":1,"minTime":0,"highWater":0,"strategy":"OVERFLOW","reservoir":20,"reservoirIncreaseInterval":300000,"reservoirIncreaseAmount":2,"reservoirIncreaseMaximum":10},"STATE_REPORT_DEFAULT":{"maxConcurrent":1,"minTime":0,"highWater":0,"strategy":"BLOCK","penalty":60000,"reservoir":60,"reservoirIncreaseInterval":300000,"reservoirIncreaseAmount":15,"reservoirIncreaseMaximum":60}},"profileMapping":{"PHYSICAL_INTERACTION_DEFAULT":"PHYSICAL_INTERACTION_DEFAULT","VOICE_INTERACTION_DEFAULT":"VOICE_INTERACTION_DEFAULT","STATE_REPORT_DEFAULT":"STATE_REPORT_DEFAULT"}}}
15 Jan 19:05:12 - [info] ***@***.*** MQTT
publish to topic
$aws/things/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/shadow/get: {}
15 Jan 19:05:12 - [info] ***@***.***
MsgRateLimiter.overrideConfig: {
  "profiles": {
    "DEFAULT": {
      "maxConcurrent": 1,
      "minTime": 1000,
      "highWater": 0,
      "strategy": "BLOCK",
      "penalty": 30000,
      "reservoir": 60,
      "reservoirIncreaseInterval": 60000,
      "reservoirIncreaseAmount": 1,
      "reservoirIncreaseMaximum": 60
    },
    "PHYSICAL_INTERACTION_DEFAULT": {
      "maxConcurrent": 1,
      "minTime": 2500,
      "highWater": 1,
      "strategy": "LEAK",
      "reservoir": 30,
      "reservoirIncreaseInterval": 900000,
      "reservoirIncreaseAmount": 1,
      "reservoirIncreaseMaximum": 15
    },
    "VOICE_INTERACTION_DEFAULT": {
      "maxConcurrent": 1,
      "minTime": 0,
      "highWater": 0,
      "strategy": "OVERFLOW",
      "reservoir": 20,
      "reservoirIncreaseInterval": 300000,
      "reservoirIncreaseAmount": 2,
      "reservoirIncreaseMaximum": 10
    },
    "STATE_REPORT_DEFAULT": {
      "maxConcurrent": 1,
      "minTime": 0,
      "highWater": 0,
      "strategy": "BLOCK",
      "penalty": 60000,
      "reservoir": 60,
      "reservoirIncreaseInterval": 300000,
      "reservoirIncreaseAmount": 15,
      "reservoirIncreaseMaximum": 60
    }
  },
  "profileMapping": {
    "PHYSICAL_INTERACTION_DEFAULT": "PHYSICAL_INTERACTION_DEFAULT",
    "VOICE_INTERACTION_DEFAULT": "VOICE_INTERACTION_DEFAULT",
    "STATE_REPORT_DEFAULT": "STATE_REPORT_DEFAULT"
  }
}
15 Jan 19:05:12 - [info] ***@***.*** MQTT
message received on topic
$aws/things/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/shadow/get/accepted:
{"state":{"reported":{"connected":false,"vsh_version":"3.0.7","nr_version":"3.0.2","devices":{"vshd-20e8f33d3bac3f85":{"friendlyName":"AAA
motion
test","template":"MOTION_SENSOR","retrievable":true}}}},"metadata":{"reported":{"connected":{"timestamp":1673805874},"vsh_version":{"timestamp":1673805435},"nr_version":{"timestamp":1673805435},"devices":{"vshd-20e8f33d3bac3f85":{"friendlyName":{"timestamp":1673464935},"template":{"timestamp":1673464935},"retrievable":{"timestamp":1673464935}}}}},"version":59,"timestamp":1673805912}
15 Jan 19:05:16 - [info] ***@***.*** MQTT
publish to topic
$aws/things/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/shadow/update:
{"state":{"reported":{"connected":true,"vsh_version":"3.0.7","nr_version":"3.0.2"}}}
15 Jan 19:05:35 - [info] ***@***.*** MQTT
publish to topic
vsh/vsht-038e8662-4e37-47d4-a85a-de7dfc39c9cf/changeReport:
{"template":"MOTION_SENSOR","endpointId":"vshd-20e8f33d3bac3f85","properties":[{"namespace":"Alexa.MotionSensor","name":"detectionState","value":"DETECTED","changed":true}],"correlationToken":"","causeType":"PHYSICAL_INTERACTION","vshVersion":"3.0.7","userIdToken":"amzn1.account.AGMCT45ZEN4ZTKV7632BTOCNMGAQ#00a5e4d44ba0cecc2296127d2e2ab6043e994abc"}


śr., 11 sty 2023 o 22:03 Cornelius Suermann ***@***.***>
napisał(a):

**Comment by csuermann** [comment]
Thanks! Everything looks as it should. I just tried to reproduce it locally (without success) and realized that your payload to change your motion sensor's state might be off.

Could you please try the payload provided as example (Node-RED > Import > Examples > node-red-contrib-virtual-smart-home > Motion Sensor)?

With the `retrievable` option turned on, this should work:
```
[{"id":"a67f98de.750138","type":"vsh-virtual-device","z":"3156dd60.366d12","name":"motion test","topic":"","connection":"5b569646351b5e75","template":"MOTION_SENSOR","retrievable":true,"passthrough":false,"diff":false,"filter":false,"filterTopic":false,"x":1570,"y":560,"wires":[[]]},{"id":"c9e0153d.a8fd58","type":"inject","z":"3156dd60.366d12","name":"motion","props":[{"p":"payload.detectionState","v":"DETECTED","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":1230,"y":560,"wires":[["a67f98de.750138"]]},{"id":"51f1a442.3442ac","type":"inject","z":"3156dd60.366d12","name":"no motion","props":[{"p":"payload.detectionState","v":"NOT_DETECTED","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","x":1240,"y":600,"wires":[["a67f98de.750138"]]},{"id":"5b569646351b5e75","type":"vsh-connection","name":"test","port":"8883","accessTokenExpiry":"1668869715652","debug":false,"backendUrl":"https://kfd5m4a21f.execute-api.eu-west-1.amazonaws.com/dev","lwaClientId":"amzn1.application-oa2-client.3f1bb07133854b078261ad43f2484c18"}]
```
